src/main/java/com/github/pjfanning/xlsx/impl/ooxml/OoXmlStrictConverterInputStream.java

```diff
@@ -0,0 +1,183 @@
+                throw new RuntimeException(e);
+            }
+            xmlConverter = null;
+        }
+        input.close();
+        output.close();
+        super.close();
+    }
+
+    private void keepReadingFromInput() throws IOException {
+        switch (readState) {
+            case BEFORE_ZIP_ENTRY:
+                readEntry();
+                break;
+            case ZIP_ENTRY_START:
+                startZipEntry();
+                break;
+            case IN_NON_XML_ENTRY:
+                readNonXmlEntry();
+                break;
+            case XML_ENTRY_START:
+                startXmlEntry();
+                break;
+            case IN_XML_ENTRY:
+                readXmlEntry();
+                break;
+        }
+    }
+
+    private void readXmlEntry() {
+        try {
+            if (!xmlConverter.convertNextElement()) {
+                xmlConverter.close();
+                xmlConverter = null;
+                readState = ReadState.BEFORE_ZIP_ENTRY;
+            }
+        } catch (XMLStreamException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void startXmlEntry() {
+        try {
+            this.xmlConverter = new OoXmlStrictConverter(OoXmlStrictConverterUtils.disableClose(input), OoXmlStrictConverterUtils.disableClose(output));
+            readState = ReadState.IN_XML_ENTRY;
+        } catch (XMLStreamException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void readNonXmlEntry() throws IOException {
+        int readCount = input.read(nonXmlEntryReadBuffer);
+        if (-1 == readCount) {
+            input.closeEntry();
+            output.closeEntry();
+            readState = ReadState.BEFORE_ZIP_ENTRY;
+            return;
+        }
+
+        if (0 != readCount) {
+            output.write(nonXmlEntryReadBuffer, 0, readCount);
+            output.flush();
+        }
+    }
+
+    private void startZipEntry() throws IOException {
+        ZipEntry newZipEntry = new ZipEntry(zipEntry.getName());
+        output.putNextEntry(newZipEntry);
+
+        if (OoXmlStrictConverterUtils.isXml(zipEntry.getName())) {
+            readState = ReadState.XML_ENTRY_START;
+            return;
+        }
+
+        readState = ReadState.IN_NON_XML_ENTRY;
+    }
+
+    private void readEntry() throws IOException {
+        zipEntry = input.getNextEntry();
+        if (zipEntry == null) {
+            readState = ReadState.END_STATE;
+            output.flush();
+            output.close();
+            return;
+        }
+        readState = ReadState.ZIP_ENTRY_START;
+    }
+
+    private enum ReadState {
+        BEFORE_ZIP_ENTRY,
+        ZIP_ENTRY_START,
+        END_STATE,
+        XML_ENTRY_START,
+        IN_NON_XML_ENTRY,
+        IN_XML_ENTRY,
+    }
+
+    private static class ByteArrayOutputStreamExposed extends ByteArrayOutputStream {
+
+        public byte[] getBytes() {
+            return buf;
```

Comment:
this is more efficient than https://docs.oracle.com/javase/7/docs/api/java/io/ByteArrayOutputStream.html#toByteArray() ?

Reply:
`toByteArray()` creates a copy. Probably not that important, performance-wise, considering we're doing IO and unzipping and zipping unnecessarily but just seemed weird to clone the array when all I need to do is read from it.